compute_endpoint/globus_compute_endpoint/boot_persistence.py

```diff
@@ -50,28 +49,29 @@ def enable_on_boot(ep_dir: pathlib.Path):
     ep_name = ep_dir.name
 
     config = get_config(ep_dir)
-    if config.detach_endpoint:
-        # config.py takes priority if it exists
-        if os.path.isfile(ep_dir / "config.py"):
-            # can't give users a nice packaged command to run here, so just tell them
-            msg = (
-                f"Persistent endpoints cannot run in detached mode. Update {ep_name}'s"
-                " config to set detach_endpoint=False and try again."
+    if not config.multi_user:
+        if config.detach_endpoint:
+            # config.py takes priority if it exists
+            if (ep_dir / "config.py").is_file():
+                # can't give users a nice packaged command to run here; just tell them
+                msg = (
+                    "Persistent endpoints cannot run in detached mode.  Update"
+                    f" {ep_name} config to set `detach_endpoint=False` and try again."
+                )
+            else:
+                # _can_ give a nice packaged command if they use yaml, though
+                msg = (
+                    "Persistent endpoints cannot run in detached mode.  Run the"
+                    f" following command to update {ep_name}'s config:"
+                    f"\n\techo 'detach_endpoint: false' >> {ep_dir / 'config.yaml'}"
+                )
+            raise ClickException(msg)
+
+        if Endpoint.check_pidfile(ep_dir)["active"]:
+            raise ClickException(
+                "Cannot enable on-boot persistence for an endpoint that is currently"
+                f" running. Stop endpoint {ep_name} and try again."
             )
```

Comment:
why not check this in MEPs too? I find it cleaner if we either always check or never check (unless there's a technical reason to do otherwise)

Reply:
Because we don't currently write a PID file in the MEP and I didn't want to go down that avenue at this time.

Reply:
(To be clear, I'm not inclined to write a PID file in the MEP precisely now.  Can follow that up at a later date.)